test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -0,0 +1,520 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using MessagePack;
+using Microsoft;
+using Microsoft.VisualStudio.Threading;
+using Nerdbank.Streams;
+using Newtonsoft.Json;
+using StreamJsonRpc;
+using Xunit;
+using Xunit.Abstractions;
+
+/// <summary>
+/// Tests the proxying of interfaces marked with <see cref="RpcMarshalableAttribute"/>.
+/// </summary>
+public abstract class MarshalableProxyTests : TestBase
```

Comment:
I replicated all the tests that are available for `IDisposable` and added some making sure that
- `DisposeOnDisconnect` is honored
- the proxies can be used
- exceptions are thrown if the interfaces are invalid for marshaling (non disposable, have events, or have properties)
- generic interfaces work